server/render.js

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const Eleventy = require('@11ty/eleventy');
+const path = require('path');
+const yaml = require('js-yaml');
+const xss = require('xss').filterXSS;
+const {EleventyRenderPlugin} = require('@11ty/eleventy');
+const {OAuth2Client} = require('google-auth-library');
+
+const ignorePatterns = [
+  'site/es/**/*',
+  'site/pt/**/*',
+  'site/ko/**/*',
+  'site/zh/**/*',
+  'site/ru/**/*',
+  'site/ja/**/*',
+  'site/fr/**/*',
+  'site/de/**/*',
+  'site/**/docs/',
+  'site/**/articles/',
+  'site/**/authors/',
+  'site/**/tags/',
+  'site/**/releases/',
+  'site/**/workbox/',
+];
+
+const CLIENT_ID =
+  '636061184119-c4pof3b538r4ibs8bt6g8qdfth958nc9.apps.googleusercontent.com';
+const authClient = new OAuth2Client(CLIENT_ID);
+
+async function verifyLogin(token) {
+  const ticket = await authClient.verifyIdToken({
+    idToken: token,
+    audience: CLIENT_ID,
+  });
+  const payload = ticket.getPayload();
+  return payload && payload['sub']; // userid
+}
+
+/**
+ * Escapes and formats frontmatter provided by user to 11ty format.
+ * @param {string} inputFm Frontmatter string
+ */
+const parseFrontmatter = inputFm => {
+  const /** @type {any} */ parsedFm = yaml.load(inputFm) || {};
+  const date = parsedFm.date?.toISOString ? parsedFm.date : new Date();
+  return {
+    layout: xss(parsedFm.layout) || 'layouts/blog-post.njk',
+    date: xss(date.toISOString().split('T')[0]),
```

Comment:
That probably doesn't need to be piped through `xss` as `Date` wouldn't produce malicious strings, no?

Reply:
right:)